apollo-router/src/json_ext.rs

```diff
@@ -852,16 +862,7 @@ impl<'de> serde::de::Visitor<'de> for FlattenVisitor {
     {
         let mut type_conditions: Vec<String> = Vec::new();
         let path = TYPE_CONDITIONS_REGEX.replace(s, |caps: &Captures| {
-            type_conditions.extend(
-                caps.name("condition")
-                    .map(|c| {
-                        c.as_str()
-                            .split(',')
-                            .map(|s| s.to_string())
-                            .collect::<Vec<_>>()
-                    })
-                    .unwrap_or_default(),
-            );
+            type_conditions.extend(extract_matched_conditions(caps).unwrap_or_default());
```

Comment:
Note that `foo|[]` is not equivalent to `foo` here (or rather, empty type conditions are not the same as no type conditions). Previously we were effectively disallowing empty type conditions entirely via the regex. If want to allow them, we should be returning `Some(vec![])` rather `None`.